modules/xfem/src/userobjects/NodeValueAtXFEMInterface.C

```diff
@@ -46,61 +43,52 @@ PointValueAtXFEMInterface::PointValueAtXFEMInterface(const InputParameters & par
 }
 
 void
-PointValueAtXFEMInterface::initialize()
+NodeValueAtXFEMInterface::initialize()
 {
   _pl = _mesh.getPointLocator();
+  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(_fe_problem.getXFEM());
+  if (_xfem == nullptr)
+    mooseError("Problem casting to XFEM in NodeValueAtXFEMInterface");
 
   const UserObject * uo =
-      &(_fe_problem.getUserObjectBase(getParam<UserObjectName>("geometric_cut_userobject")));
+      &(_fe_problem.getUserObjectBase(getParam<UserObjectName>("interface_mesh_cut_userobject")));
 
-  if (dynamic_cast<const LineSegmentCutSetUserObject *>(uo) == nullptr)
-    mooseError("UserObject casting to GeometricCutUserObject in XFEMSingleVariableConstraint");
-
-  _geo_cut = dynamic_cast<const LineSegmentCutSetUserObject *>(uo);
-
-  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(_fe_problem.getXFEM());
-  if (_xfem == nullptr)
-    mooseError("Problem casting to XFEM in PointValueAtXFEMInterface");
+  if (dynamic_cast<const InterfaceMeshCutUserObjectBase *>(uo) == nullptr)
+    mooseError("UserObject casting to InterfaceMeshCutUserObjectBase in NodeValueAtXFEMInterface");
 
-  _elem_pairs = _xfem->getXFEMCutElemPairs(_xfem->getGeometricCutID(_geo_cut));
+  _mesh_cut = dynamic_cast<const InterfaceMeshCutUserObjectBase *>(uo);
+  _elem_pairs = _xfem->getXFEMCutElemPairs(_xfem->getGeometricCutID(_mesh_cut));
 }
 
 void
-PointValueAtXFEMInterface::execute()
+NodeValueAtXFEMInterface::execute()
 {
   _values_positive_level_set_side.clear();
   _values_negative_level_set_side.clear();
   _grad_values_positive_level_set_side.clear();
   _grad_values_negative_level_set_side.clear();
-  _points.clear();
+  _nodes.clear();
 
-  std::vector<Real> cut_data = _geo_cut->getCutData();
+  std::shared_ptr<MeshBase> cutter_mesh = _mesh_cut->getCutterMesh();
 
-  const int line_cut_data_len = 6;
-  for (unsigned int i = 0; i < cut_data.size() / line_cut_data_len; ++i)
-  {
-    _points.push_back(
-        Point(cut_data[i * line_cut_data_len + 0], cut_data[i * line_cut_data_len + 1]));
-    if (i == cut_data.size() / line_cut_data_len - 1)
-      _points.push_back(
-          Point(cut_data[i * line_cut_data_len + 2], cut_data[i * line_cut_data_len + 3]));
-  }
+  for (const auto & node : cutter_mesh->node_ptr_range())
+    _nodes.push_back(*node);
 
-  BoundingBox bbox = _mesh.getInflatedProcessorBoundingBox();
+  _pl->enable_out_of_mesh_mode();
 
   std::vector<Point> point_vec(1);
 
-  for (MooseIndex(_points) i = 0; i < _points.size(); ++i)
+  for (const auto & node : cutter_mesh->node_ptr_range())
   {
-    Point p = _points[i];
+    unsigned int i = node->id();
 
-    if (bbox.contains_point(p))
+    if ((*_pl)(*node) != nullptr)
     {
-      const Elem * elem = getElemContainingPoint(p, true);
+      const Elem * elem = getElemContainingPoint(*node, true);
```

Comment:
```suggestion
      const Elem * elem = getElemContainingPoint(*node, /*positive_level_set = */true);
```
It's better to a tiny inline comment to give future developers some hints.

Reply:
done.